### DISCUSSION TITLE
Special characters in markdown

### DISCUSSION THREAD

**Discussion by syeda-git** [squidfunk/mkdocs-material#5400] [Q&A]
Is it possible to report in log or stop building HTML when there are special characters (< and > in this case) in markdown?

If there are < and > characters in HTML files, the translation engine fails to parse such files.

For example, this para has two strings with < and > characters. I would like Material for MkDocs to report in log or stop building when such characters are found in Markdown.

# Product Overview

HCL Digital Experience (DX) is the trusted solution for an enterprise's digital experience at the intersection of processes, content, and applications. HCL DX uses <sometext1> and <sometext2>. HCL DX is architected to build more than just websites. It is built Cloud Native to serve your customers, employees, and partners and to deliver the functionality and stability that they need and expect.

Thanks...
Syed

**Comment by squidfunk** [comment]
The Markdown parser escapes them as `&lt;` and `&gt;`, so I'm not sure why this is necessary? Other than that, you could write a [hook](https://www.mkdocs.org/user-guide/configuration/#hooks) and inspect `on_page_markdown` to check for those characters and fail the build. Similar to:

``` python
def on_page_markdown(markdown, *, page, config, files):
    if "<" in markdown or ">" in markdown:
        # Do something ...
```

Note that using HTML in Markdown is valid, so the hook above would run when it sees any HTML tag.
